### DISCUSSION TITLE
[home page] make the tab/pills sticky

### DISCUSSION THREAD

**Discussion by mistercrunch** [apache/superset#19198] [Ideas]
<img width="520" alt="Screen Shot 2021-02-23 at 10 26 10 AM" src="https://user-images.githubusercontent.com/487433/108889803-a1e1cd80-75c1-11eb-903c-74ef4b0af593.png">

I take the time to set my favorite dashboards and charts, but the home page won't remember that I prefer "Favorites" over "Mine". Let's make that sticky. Let's also create a nice simple abstraction for things to be sticky. Maybe that becomes a `sticky` React prop for our Tab/Pills component in our design system (?)

**Comment by nytai** [comment]
As in save the preference to local storage or in metadata db?

**Comment by nytai** [comment]
I would be nice to create a framework where this is saved in the metadata db instead of local storage

**Comment by junlincc** [comment]
my first reaction is local storage, that's how we set in Explore, but after thinking through, it might make more sense to save it in metadata db. @nytai

**Comment by nytai** [comment]
Let's just work with a good abstraction that should allow us to switch over to a different storage mechanism later on. As the UI gets more complex and we create more of these sticky options, it would be a shame for all of that to be lost once someone upgrades to a new computer which is fairly common in corporate settings.

**Comment by ktmud** [comment]
+1 on good abstraction. Another benefit of saving things like this to db is cross-platform/browser consistency.
